pkgs/by-name/sw/swaymux/package.nix

```diff
@@ -0,0 +1,40 @@
+{ cmake
+, fetchFromGitea
+, lib
+, nlohmann_json
+, qt6
+, qtbase
+, qtwayland
+, stdenv
+, wrapQtAppsHook
+}:
+stdenv.mkDerivation (finalAttrs: {
+  version = "1.0";
+  pname = "swaymux";
+
+  src = fetchFromGitea {
+    rev = "v${finalAttrs.version}";
+    domain = "git.grimmauld.de";
+    owner = "Grimmauld";
+    repo = "swaymux";
+    hash = "sha256-M85pqfYnYeVPTZXKtjg/ks5LUl3u2onG9Nfn8Xs+BSA=";
+  };
+
+  buildInputs = [ qt6.qtwayland cmake nlohmann_json qtbase];
+  nativeBuildInputs = [ cmake qt6.wrapQtAppsHook ];
+
+  doCheck = true;
+
+  meta = with lib; {
```

Comment:
Optional: You can easily get rid of `with lib;` here.

Reply:
yes, but i see no reason; its about 50/50 whether `with lib;` is used or not, and i believe it to be more readable keeping that in.

Reply:
This is why I marked this as optional. Many maintainers are removing `with lib;` when they are prefixing a set because it's confusing. It's also make the work of static analysers more complex. To many of use, the use of `with <>;` should be mostly used in front of lists only.

Reply:
that is fair i guess, but seeing as this is optional i'll keep it the way i prefer on my package. Static analyzers might have to work harder, but if i can personally read it faster like this then why would i make it easy for the machine and difficult for myself.